uaclient/entitlements/fips.py

```diff
@@ -154,6 +155,56 @@ def conditional_packages(self):
 
         return FIPS_CONDITIONAL_PACKAGES.get(series, [])
 
+    def prompt_if_kernel_downgrade(
+        self,
+        assume_yes: bool = False,
+    ) -> bool:
+        """Check if installing a FIPS kernel will downgrade the kernel
+        and prompt for confirmation if it will.
+        """
+        # Prior to installing packages, check if the kernel is being downgraded
+        # and if so verify that the user wants to continue
+        our_full_kernel_str = (
+            system.get_kernel_info().proc_version_signature_version
+        )
+        if our_full_kernel_str is None:
+            LOG.warning("Cannot gather kernel information")
+            return False
+        our_m = re.search(
+            r"(?P<kernel_version>\d+\.\d+\.\d+)", our_full_kernel_str
+        )
+        fips_kernel_version_str = apt.get_pkg_candidate_version("linux-fips")
+        if our_m is not None and fips_kernel_version_str is not None:
+            our_kernel_version_str = our_m.group("kernel_version")
+            LOG.debug(
+                "Kernel information: cur={cur} and fips={fips}".format(
```

Comment:
For the logs, we are using the string interpolation strategy as you can see here:
https://github.com/canonical/ubuntu-pro-client/blob/main/uaclient/entitlements/base.py#L349

Note there is nothing wrong with the logs you have written, but I think we just change them just be consistent with the other logs we have already defined

Reply:
I will make the update to use that format but I will note that string interpolation can be significantly harder to read, especially when there are multiple variables to consider.  I would like to see us migrate to format strings e.g. `f"{Kernel information: {our_full_kernel_str=} and {fips_kernel_version_str}"` but that would be a much larger discussion/refactor and code consistency is important.